scripts/nbconverted/explore_HighWeight_genes.py

```diff
@@ -0,0 +1,242 @@
+    add_metadata_pos = pd.DataFrame({node_name: [node_mean, node_std, pos_threshold]})
+    HW_metadata_df = pd.concat([HW_metadata_df, add_metadata_pos], axis=1)
+    
+    # Add high weight negative genes
+    node_name = 'Sig'+node+'neg'
+    
+    add_neg = pd.DataFrame({node_name: list(hw_neg_genes)})
+    HW_df = pd.concat([HW_df, add_neg], axis=1)
+    
+    # Add metadata for negative node
+    add_metadata_neg = pd.DataFrame({node_name: [node_mean, node_std, neg_threshold]})
+    HW_metadata_df = pd.concat([HW_metadata_df, add_metadata_neg], axis=1)
+
+HW_df = HW_df.T
+HW_metadata_df = HW_metadata_df.T
+
+
+# In[8]:
+
+
+HW_df.head()
+
+
+# In[9]:
+
+
+# Dataframe with the mean, std, threshold used per feature to determine high weight gene sets
+HW_metadata_df.columns = ["mean", "std", "threshold"]
+HW_metadata_df.head()
+
+
+# In[10]:
+
+
+# Create a table (feature x gene set A)
+# Each cell will contain the distance between the weight and the mean
+
+Weights_pos_neg_df = pd.DataFrame()
+
+for i in range(num_nodes):
+    node_mean = means.iloc[i]
+    
+    node = str(i+1)
+    node_name = 'Sig'+node+'pos'
+    
+    gene_weight = weight[i+1]
+    
+    # Add high weight positive genes
+    add_pos = pd.DataFrame({node_name: gene_weight})
+    Weights_pos_neg_df = pd.concat([Weights_pos_neg_df, add_pos], axis=1)
+    
+    # Add high weight negative genes
+    node_name = 'Sig'+node+'neg'
+    
+    add_neg = pd.DataFrame({node_name: gene_weight})
+    Weights_pos_neg_df = pd.concat([Weights_pos_neg_df, add_neg], axis=1)
+    
+    
+Weights_pos_neg_df = Weights_pos_neg_df.T
+Weights_pos_neg_df.head()
+
+
+# In[11]:
+
+
+# Compare the weights for geneset A with threshold
+geneSetA_ls = [l[0] for l in geneSetA.values.tolist()]
+Weight_A = Weights_pos_neg_df[geneSetA_ls]
+
+Weight_A.head()
+
+
+# In[12]:
+
+
+sns.distplot(Weight_A.iloc[0])
+
+
+# In[13]:
+
+
+# Compare the weights for geneset B with threshold
+geneSetB_ls = [l[0] for l in geneSetB.values.tolist()]
+Weight_B = Weights_pos_neg_df[geneSetB_ls]
+
+Weight_B.head()
+
+
+# In[14]:
+
+
+sns.distplot(Weight_B.iloc[0])
+
+
+# In[15]:
+
+
+# What is the overlap between the high weight genes and gene sets A and B?
+num_A = geneSetA.shape[0]
+num_B = geneSetB.shape[0]
+
+num_features_pos_neg = HW_df.shape[0]
+
+percent_overlap = pd.DataFrame({'feature': [], 
+                                'percent in A': [],
+                                'percent in B': []
+                               })
+
+for i in range(num_features_pos_neg):
+#for i in range(2):
+    row = HW_df.iloc[i]
+    percent_in_A = (row.isin(geneSetA_ls).sum()/num_A)*100
+    percent_in_B = (row.isin(geneSetB_ls).sum()/num_B)*100
+    
+    add = pd.DataFrame({'feature': [HW_df.index[i]],
+                        'percent in A': [percent_in_A],
+                        'percent in B': [percent_in_B]
+                       })
+    percent_overlap = percent_overlap.append(add)
+
+percent_overlap
```

Comment:
Why do we care about the percent overlap? If the geneset is in the feature then we can say something about that geneset or that feature? or is this to nominate/prioritize features?

Also, do you want to sort this to highlight the nodes that have the highest percent overlap?

Reply:
This was just a very first pass at exploring the data.  Based on the results of the interpolation analysis we were NOT seeing a discrepancy between the ability for VAE vs PCA vs No compression models to predict intermediate gene expression levels as gene set A is becoming activated and gene set B is getting upregulated.  So we first were wondering if genes and A and B were being captured by our method.  I guess I could've made it binary to look for presence or absence, but I didn't think of that at the time.  Instead I just looked at percentages which gave me information about "how much" of the genes in A or B were captured in addition to their presence.  But there is no ranking being done here.